### ISSUE TITLE
`Group by` and `left join using` gives a "column reference is ambiguous

### ISSUE THREAD

**Issue by Yamakaky** [sqlc-dev/sqlc#1425]
### Version

1.12.0

### What happened?

When using `join ... using (key) group by key`, sqlc seems to think `key` references the columns in the joined tables instead of the synthetic `key` column generated by `using`. `where` and `having` doesn't seem to have this problem, they work as expected.

This seem to be a regression since I have no problem with this kind of query in the 1.10.0 I use.

### Relevant log output

```shell
sqlc generate failed.

# package db
query.sql:7:46: column reference "fk" is ambiguous
```

### Database schema

```sql
create table t1 (
        fk integer not null unique
);
create table t2 (
        fk integer not null references t1(fk)
);
```

### SQL queries

```sql
select * from t1 join t2 using (fk) group by fk;
```

### Playground URL

https://play.sqlc.dev/p/f9717de599d83359e7f1fdf15bbe7e3ed9cb57ce7c6b4c9dd6576a6445b140cd

### What database engines are you using?

PostgreSQL

**Comment by sbres** [comment]
Your problem comes from your query, It's not valid.

The SQLengine does not know what table you want to use for the group by.

I was able to run your example by specifying the table form where to group by 
```
select * from t1 join t2 using (fk) group by t1.fk;
```
I've attached your setup in playground
https://play.sqlc.dev/p/aaaa171855439c5253e3c885154d0f6a23b2953bb1625fc9a02b07b2ffe941c4

**Comment by Yamakaky** [comment]
I just tested this in postgresql, it works.

Updated the schema to add unique constraint.

**Comment by Yamakaky** [comment]
Any news on that ?

From [postgresql doc](https://www.postgresql.org/docs/current/queries-table-expressions.html): "Furthermore, the output of JOIN USING suppresses redundant columns: there is no need to print both of the matched columns, since they must have equal values. While JOIN ON produces all columns from T1 followed by all columns from T2, JOIN USING produces one output column for each of the listed column pairs (in the listed order), followed by any remaining columns from T1, followed by any remaining columns from T2."

**Comment by BaronBonet** [comment]
I'm also running into this issue, any update?

**Comment by kyleconroy** [comment]
This is fixed in [v1.23.0](https://docs.sqlc.dev/en/latest/reference/changelog.html#v1-23-0) by enabling the [database-backed query analyzer](https://docs.sqlc.dev/en/latest/howto/generate.html#enhanced-analysis-with-managed-databases). We added a [test case](https://github.com/sqlc-dev/sqlc/tree/main/internal/endtoend/testdata/join_using/postgresql/pgx) for this issue so it won’t break in the future.

You can play around with the [working example on the playground](https://play.sqlc.dev/p/6bb79f58dd07a6f234ab7ff4b23c799ea60abee8243441edb95f2cbffc6d530a)
